Refactor: Standardize Movie entity and UI models

skaik-mo opened the pull request:
This commit refactors the `Movie` entity and related UI models to use consistent field names (`name` instead of `title`, `overview` instead of `description`).

It also introduces an `orEmpty()` extension function to handle nullable string and float values more cleanly across different presentation modules.

Specific changes include:
- Renamed `title` to `name` and `description` to `overview` in `Movie` entity and associated UI models (`MovieUi`, `HomeUiModel`, `PosterUiState`, `PopularMediaUiModel`, `HistoryUiModel`, `RatedPoster`).
- Replaced direct null checks and default value assignments with `orEmpty()` extension for various string and float properties in mappers and UI models.
- Added `orEmpty()` extension function to `presentation/home`, `presentation/profile`, and `presentation/explore` modules.
- Updated `toFormattedDate` to accept nullable `LocalDate` and `toFormattedDuration` to accept nullable `Int`.
- Removed unused `recentReleasedAt` and `upcomingAt` fields from `Movie` entity and updated related use case tests.
- Adjusted mappers in `repository/media` and presentation modules to align with the new `Movie` entity structure and use `orEmpty()`.
- Updated `SeriesDetailsViewModel` to use `SeriesUi.fromEntity` directly.

skaik-mo committed:
Refactor: Standardize Movie entity and UI models

This commit refactors the `Movie` entity and related UI models to use consistent field names (`name` instead of `title`, `overview` instead of `description`).

It also introduces an `orEmpty()` extension function to handle nullable string and float values more cleanly across different presentation modules.

Specific changes include:
- Renamed `title` to `name` and `description` to `overview` in `Movie` entity and associated UI models (`MovieUi`, `HomeUiModel`, `PosterUiState`, `PopularMediaUiModel`, `HistoryUiModel`, `RatedPoster`).
- Replaced direct null checks and default value assignments with `orEmpty()` extension for various string and float properties in mappers and UI models.
- Added `orEmpty()` extension function to `presentation/home`, `presentation/profile`, and `presentation/explore` modules.
- Updated `toFormattedDate` to accept nullable `LocalDate` and `toFormattedDuration` to accept nullable `Int`.
- Removed unused `recentReleasedAt` and `upcomingAt` fields from `Movie` entity and updated related use case tests.
- Adjusted mappers in `repository/media` and presentation modules to align with the new `Movie` entity structure and use `orEmpty()`.
- Updated `SeriesDetailsViewModel` to use `SeriesUi.fromEntity` directly.

poula22 reviewed (dismissed):
Great job👏

skaik-mo committed:
Refactor: Standardize MovieRepository method names

This commit refactors the method names in `MoviesRepository` and its implementations/usages to be more concise and consistent.

Key changes:
- `getMovie...` methods are now `get...` (e.g., `getMovieDetails` -> `getDetails`)
- `searchMovieByName` is now `getByName`
- `addRating`'s parameter `ratingValue` is now `rating`
- `clearMovieCache` is now `clearAll`
- `clearMovieCacheWithOutRecentViewed` is now `clearExceptRecentlyViewed`
- `clearMovieGenres` is now `clearGenres`
- `deleteMovieById` is now `deleteById`
- `deleteMovieRating` is now `deleteRating`

This improves readability and maintainability of the codebase.

skaik-mo committed:
Refactor: Rename MoviesRepository to MovieRepository and update related use cases

This commit renames `MoviesRepository` to `MovieRepository` for consistency and updates all related use cases to reflect this change.

Additionally:
- `GetPopularityMoviesUseCase` is renamed to `GetPopularMoviesUseCase`.
- `GetRecommendedMovieUseCase` is renamed to `GetRecommendedMoviesUseCase`.
- `SearchMovieByNameUseCase` is renamed to `GetMovieByNameUseCase`.
- Methods in `ClearMoviesCacheUseCase` are renamed for clarity:
    - `clearMovieCache` to `clearAll`
    - `clearMovieCacheWithOutRecentViewed` to `clearExceptRecentlyViewed`
    - `clearRecentlyViewedMovies` to `clearRecentlyViewed`
- Parameter names in some use case `invoke` methods are updated for clarity (e.g., `ratingValue` to `rating`, `pageNumber` to `page`).

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop: (22 commits)
  unify onError callback for series details viewmodel
  use reference function and unify onError handlers
  clean up
  clean up
  move routes to rotes package
  move routes to rotes package
  move routes to rotes package
  move routes to rotes package
  add mappers
  add map error to string resource utility
  refactor retry action
  refactor reviews view model to get rid of string resources in the view model
  refactor seasons view model
  refactor series details view model
  refactor recommended series view model
  refactor recommended movies view model
  refactor recommended movies view model
  refactor movie details view model
  refactor gallary view model
  refactor cast details view model
  ...

# Conflicts:
#	presentation/details/src/main/java/com/giraffe/presentation/details/model/MovieUi.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/model/SeasonUi.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/model/SeriesUi.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/moviedetails/MovieDetailsViewModel.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/recommended/movie/RecommendedMoviesViewModel.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/reviewScreen/ReviewsViewModel.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/seriesdetails/SeriesDetailsViewModel.kt

mohamedebrahem13 commented on presentation/details/src/main/java/com/giraffe/presentation/details/utils/ConvertFormatted.kt:
```diff
@@ -1,9 +1,11 @@
 
-fun Int.toFormattedDuration(): String {
+fun Int?.toFormattedDuration(): String {
+    if (this == null || this < 0) return ""
```
use ?.let instead of check for null here

mohamedebrahem13 commented on presentation/details/src/main/java/com/giraffe/presentation/details/utils/ConvertFormatted.kt:
```diff
@@ -12,11 +14,12 @@ fun Int.toFormattedDuration(): String {
-    val inputFormat = SimpleDateFormat("yyyy-M-dd", Locale.ENGLISH)
-    val date = inputFormat.parse(this) ?: return this
+fun LocalDate?.toFormattedDate(): String {
+    if (this == null) return ""
```
here also use ?.let instead of check for null here

mohannadahmed00 commented on domain/media/src/main/java/com/giraffe/media/movie/usecase/GetMovieByNameUseCase.kt:
```diff
@@ -1,15 +1,15 @@
 
-class SearchMovieByNameUseCase @Inject constructor(
-    private val repository: MoviesRepository
+class GetMovieByNameUseCase @Inject constructor(
```
i think it should be Get**Movies**ByNameUseCase because you return List<Movie>

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/presentation/details/model/MovieUi.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/presentation/details/screens/moviedetails/MovieDetailsScreen.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/presentation/details/screens/moviedetails/MovieDetailsViewModel.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/presentation/details/screens/recommended/movie/RecommendedMoviesViewModel.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/details/src/main/java/com/giraffe/presentation/details/screens/reviewScreen/ReviewsViewModel.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/search/SearchViewModel.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/screen/searchresult/SearchResultViewModel.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/util/Extention.kt:
```diff
@@ -8,4 +8,6 @@ fun SearchTab.toTitle(context: Context) = when (this) {
\ No newline at end of file
+}
+
+fun Any?.orEmpty(): String = this?.toString() ?: ""
```
اه الكود المقرف ده!!!!!

mohannadahmed00 commented on presentation/explore/src/main/java/com/giraffe/explore/util/Mapper.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/home/src/main/java/com/giraffe/presentation/home/screen/show_more/ShowMoreStrategy.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/home/src/main/java/com/giraffe/presentation/home/utils/Extensions.kt:
```diff
@@ -0,0 +1,3 @@
+package com.giraffe.presentation.home.utils
+
+fun Any?.orEmpty(): String = this?.toString() ?: ""
```
مفيش فايدة!!! 🙄

mohannadahmed00 commented on presentation/home/src/main/java/com/giraffe/presentation/home/utils/Mapper.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/profile/src/main/java/com/giraffe/presentation/profile/model/SwipeablePoster.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on presentation/profile/src/main/java/com/giraffe/presentation/profile/utils/Mapper.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on repository/media/src/main/java/com/giraffe/media/collections/mapper/Mapper.kt:
it's out of your scope

skaik-mo replied:
I updated the domain layer, so anything depending on it will be affected.

mohannadahmed00 commented on repository/media/src/main/java/com/giraffe/media/movie/MovieRepositoryImpl.kt:
```diff
@@ -24,78 +24,78 @@ import kotlinx.coroutines.flow.map
-        val requestBody = RatingRequest(value = ratingValue)
-        remote.addRating(movieId, requestBody)
+class MovieRepositoryImpl @Inject constructor(
+    private val movieLocal: MoviesLocalDataSource,
```
i think local or localDatasource is better

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop: (85 commits)
  solve request changes
  resolve failed test case
  add .orempty to nallable attributes
  refactor test cases
  reformat file
  rename file to user dto and make all values nullable
  delete unused interface
  create file for dummy data
  solve request changes
  change explore keyword to search keyword
  change repo implementaion name to search keyword rep
  update test cases and create dummy file
  change injected repo name
  import missing r
  change keyword repo to search repo
  change delete keyword use case to delete search keyword
  delete unused file
  Refactor: Enhance NoInternetException handling in profile ViewModels
  Rename Effect states and remove unused error message
  Add Keep and Serializable annotations to ShowMoreSectionType
  ...

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/repositoryModule.kt
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/screen/search/SearchViewModel.kt
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/screen/searchresult/SearchResultViewModel.kt
#	presentation/profile/src/main/java/com/giraffe/presentation/profile/model/SwipeablePoster.kt
#	presentation/profile/src/main/java/com/giraffe/presentation/profile/utils/Extensions.kt
#	presentation/profile/src/main/java/com/giraffe/presentation/profile/utils/Mapper.kt

skaik-mo committed:
Refactor: Improve null safety in ConvertFormatted.kt

This commit refactors the `toFormattedDuration` and `toFormattedDate` functions in `ConvertFormatted.kt` to improve null safety by using the safe call operator (`?.let`).

The `toFormattedDuration` function now returns an empty string if the input integer is null or less than or equal to zero.
The `toFormattedDate` function now returns an empty string if the input `LocalDate` is null.

skaik-mo committed:
Refactor: Rename `GetMovieByNameUseCase` to `GetMoviesByNameUseCase`

The use case `GetMovieByNameUseCase` has been renamed to `GetMoviesByNameUseCase` to better reflect its functionality of fetching a list of movies by name.

This change also updates the corresponding ViewModel and test class to use the new name.

skaik-mo committed:
Refactor: Make `youtubeVideoId` in `Movie` non-nullable

The `youtubeVideoId` property in the `Movie` entity is now non-nullable.
This change simplifies handling of YouTube video IDs throughout the codebase by ensuring a value is always present, defaulting to an empty string if necessary.

Affected mappers have been updated to reflect this change:
- `MovieEntity.toDomain()`: Now uses `youtubeVideoId.orEmpty()` to provide a default empty string.
- `Movie.toEntity()`: Now uses `youtubeVideoId` directly as it's guaranteed to be non-null.
- `MovieResult.toDomain()`: Now provides an empty string for `youtubeVideoId`.
- `Movie.toUi()`: Now uses `youtubeVideoId` directly.

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop:
  add interactions and effects to gallary screen
  remove unneeded base screen
  add keep route in proguard rules
  clean up CastDetailsScreen
  clean up
  set base & utils
  add show toast utility & throwable to string resource
  use base screen to handle app bar & loaading state & no internet state
  Implement loading and no internet states in search result screen
  Implement loading and no internet states in discover screen
  Implement loading and no internet states in search screen
  Refactor: Move NothingFound composable to searchresult package
  Refactor: Rename `error` to `onError` in `BasePagingSource`
  Add showToast extension function for Context
  Implement `EffectListener` to collect Flow events
  Refactor: Organize Explore components
  Add Explore screen string resources

# Conflicts:
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/screen/searchresult/SearchResultViewModel.kt
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/util/Extention.kt

skaik-mo committed:
Merge branch 'develop' into refactor/domain-movie

* develop: (23 commits)
  Refactor: Handle no internet and loading states in SeasonsScreen
  Refactor: Rename `EventListener` to `EffectListener`
  Add retry functionality to NoInternetScreen
  feat: manage bottom bar visibility in match module
  rename back navigation effect
  remove loading state in match pager screen
  refactor series details screen
  handle discover screen and remove unused line in view model
  handle No internet connection when select Series after Search
  Refactor match navigation and add loading delay constant
  refactor movie details view model
  refactor movie details screen
  add movie details screen bottom sheets
  add reference for mapping and remove unnecessary code
  Update loading state in match pager
  add match pager route and update navigation
  fix naming and remove redundant code
  handle genres display and date in show more screen
  resolve conflicts between presentation and domain
  refactor home view model ,effects , naming and screen content
  ...

# Conflicts:
#	app/src/main/java/com/giraffe/cineverseapp/di/presentationModule.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/moviedetails/MovieDetailsScreen.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/screens/moviedetails/MovieDetailsViewModel.kt
#	presentation/details/src/main/java/com/giraffe/presentation/details/utils/mappers.kt
#	presentation/explore/src/main/java/com/giraffe/presentation/explore/screen/searchresult/SearchResultViewModel.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/show_more/ShowMoreState.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/screen/show_more/ShowMoreStrategy.kt
#	presentation/home/src/main/java/com/giraffe/presentation/home/utils/Mapper.kt
